### PR TITLE
rec: fix client remotes count when using proxy protocol

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

All other remote counts use the address substituted by the ProxyProtocol, so `remote` should as well.

In an ideal world we should also test  the other `remote` counts produced by https://docs.powerdns.com/recursor/http-api/endpoint-jsonstat.html#get--jsonstat

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [X] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/recursordist/rec-tcp.cc [hunk 1]
```diff
@@ -565,6 +565,9 @@ static void handleRunningTCPQuestion(int fileDesc, FDMultiplexer::funcparam_t& v
           ++iter->second.stats.netmaskMatches;
         }
       }
+      if (t_remotes) {
+        t_remotes->push_back(conn->d_source);
```
Comment (rgacogne) [inline pdns/recursordist/rec-tcp.cc:569]: Am I missing something or are we no longer recording the source over TCP when the proxy protocol is not enabled?

File: regression-tests.recursor-dnssec/test_ProxyProtocol.py [hunk 2]
```diff
@@ -134,7 +139,29 @@ class ProxyProtocolAllowedRecursorTest(ProxyProtocolRecursorTest):
...
+        headers = {'x-api-key': self._apiKey}
+        url = 'http://127.0.0.1:' + str(self._wsPort) + '/jsonstat?command=get-remote-ring&name=remotes'
+        r = requests.get(url, headers=headers, timeout=self._wsTimeout)
+        self.assertTrue(r)
+        self.assertEqual(r.status_code, 200)
+        self.assertTrue(r.json())
+        content = r.json()
+        # We allow all kind of entries, but 127.0.0.1 must have the given value, due to the
+        # testLocalProxyProtocol test, which actually does not set a source address.  If we see a
+        # higher value than expected, some ProxyProtocll clients were accounted as 127.0.0.1, which
```
Comment (rgacogne) [inline regression-tests.recursor-dnssec/test_ProxyProtocol.py:160]:
```suggestion
        # higher value than expected, some Proxy Protocol clients were accounted as 127.0.0.1, which
```

File: regression-tests.recursor-dnssec/test_ProxyProtocol.py [hunk 3]
```diff
@@ -134,7 +139,29 @@ class ProxyProtocolAllowedRecursorTest(ProxyProtocolRecursorTest):
...
+        # is not right as all other tests set a source addresses other than 127.0.0.1
```
Comment (rgacogne) [inline regression-tests.recursor-dnssec/test_ProxyProtocol.py:161]:
```suggestion
        # is not right as all other tests set a source address other than 127.0.0.1
```

File: pdns/recursordist/rec-tcp.cc (hunk 1 above)
Comment (omoerbeek) [reply pdns/recursordist/rec-tcp.cc:569]: Ugh, yes. I'll fix that

File: regression-tests.recursor-dnssec/test_ProxyProtocol.py [hunk 4]
```diff
@@ -134,7 +139,29 @@ class ProxyProtocolAllowedRecursorTest(ProxyProtocolRecursorTest):
...
+        url = 'http://127.0.0.1:' + str(self._wsPort) + '/jsonstat?command=get-remote-ring&name=remotes'
+        r = requests.get(url, headers=headers, timeout=self._wsTimeout)
+        self.assertTrue(r)
+        self.assertEqual(r.status_code, 200)
+        self.assertTrue(r.json())
+        content = r.json()
+        # We allow all kind of entries, but 127.0.0.1 must have the given value, due to the
+        # testLocalProxyProtocol test, which actually does not set a source address.  If we see a
+        # higher value than expected, some ProxyProtocol clients were accounted as 127.0.0.1, which
+        # is not right as all other tests set a source addres other than 127.0.0.1
```
Comment (rgacogne) [inline regression-tests.recursor-dnssec/test_ProxyProtocol.py:161]:
```suggestion
        # is not right as all other tests set a source address other than 127.0.0.1
```

**Review by rgacogne — APPROVED** [review]

**Pull request merged**